### DISCUSSION TITLE
Break out Lucid V2 compiler into a standalone module

### DISCUSSION THREAD

**Discussion by nishtahir** [alchitry/Alchitry-Labs-V2#2] [Ideas]
Hi, 

This is something I looked at doing briefly with V1 but found it a bit difficult due to tight coupling with the editor. Since this is a full rewrite, It might be worth considering to facilitate building other tools around the language.

**Comment by MatthewVD** [comment]
I would like to re-iterate this point with more than just an up arrow. This to me is my biggest issue with the current system and I would love to see this! If the executable could recompile / flash my alchitry cu from the command line somehow I would be extremely grateful!

**Comment by alchitry** [comment]
Would you want a fully standalone separate program or would having a robust command line interface be sufficient for your use case?

**Reply by nishtahir** [reply]
I think splitting it out into its own module would be the way to do it. if my goal was to include this in my CI pipeline, not having to download the entire editor would be ideal.

The compiler could be a shared module that the CLI and Editor could consume for example.

**Reply by MatthewVD** [reply]
A robust command line interface is sufficient for my use case. I would require some functionality for recompiling the lucid project, and then flashing it.

I have a rather strange setup with up to eight alchitry cu dev boards connected to a single computer. When I make a firmware change you can imagine that there is quite a juggling act to unplug them all and flash the updates one at a time. Additionally I don't always have a display connected to the main computer so if I could do this over ssh all the better.

I wouldn't expect you to make changes to accommodate my strange setup - but I think many would appreciate a robust command line interface that could facilitate some amount of automation!

**Reply by alchitry** [reply]
If you haven't seen it already, I wrote a blog post about the current CLI available now. 
https://alchitry.com/news/alchitry-labs-v2-command-line-interface

You can specify a board number when flashing them. I believe the numbers should be stable but I haven't done extensive testing. If this is an issue, I can look into trying to find some unique identifier for each board.
